gnucash/report/commodity-utilities.scm

```diff
@@ -173,16 +173,16 @@
                          (cons (list txn-date (/ new-domestic new-foreign)) result)))))
 
            (else
-            (warn "gnc:get-commodity-totalavg-prices: "
-                  "Sorry, currency exchange not yet implemented:"
-                  (gnc:monetary->string
-                   (gnc:make-gnc-monetary txn-comm value-amt))
-                  " (buying "
-                  (gnc:monetary->string
-                   (gnc:make-gnc-monetary price-commodity share-amt))
-                  ") =? "
-                  (gnc:monetary->string
-                   (gnc:make-gnc-monetary report-currency 0)))
+            ;; (warn "gnc:get-commodity-totalavg-prices: "
+            ;;       "Sorry, currency exchange not yet implemented:"
+            ;;       (gnc:monetary->string
+            ;;        (gnc:make-gnc-monetary txn-comm value-amt))
+            ;;       " (buying "
+            ;;       (gnc:monetary->string
+            ;;        (gnc:make-gnc-monetary price-commodity share-amt))
+            ;;       ") =? "
+            ;;       (gnc:monetary->string
+            ;;        (gnc:make-gnc-monetary report-currency 0)))
```

Comment:
Why are you removing these warnings? What has it got to do with caching the splits?

Reply:
These warnings will slow down the scanning of Forex splits. If the user chooses a multi date report eg net worth line chart spanning 10 years at daily resolution, the warnings will be shown 3650 times give or take.

Reply:
So set a flag and print the error once. Arguably it should abort the report because it will be wrong, but it might be friendlier to print a message on the report.

Regardless it should be its own commit.

Reply:
ok it'll be tricky because it'll be the _last_ report-date which captures the most splits which matters the most.

Reply:
My thoughts about showing the warnings once per report: we'd need to augment the case-exchange-date-fn to accept a "show-warnings?" optional argument which scans the splits and dumps any warnings. This would need to be called by reports once per report render.

Reply:
solved.